### PR TITLE
Add support for data logging

### PR DESCRIPTION
**Pull request by microbit-robert**

### CODE REVIEW

File: src/jshal.js [hunk 1]
```diff
@@ -215,4 +215,31 @@ mergeInto(LibraryManager.library, {
   mp_js_hal_audio_is_expression_active: function () {
     return board.audio.isSoundExpressionActive();
   },
+
+  mp_js_hal_log_delete: function (full_erase) {
+    board.dataLogging.delete();
+  },
+
+  mp_js_hal_log_set_mirroring: function (serial) {
+    board.dataLogging.delete();
```
Comment (microbit-matt-hillsdon) [inline src/jshal.js]: Is this right?

File: src/board/sensors.ts [hunk 2]
```diff
@@ -123,3 +123,38 @@ export class EnumSensor extends Sensor {
     };
   }
 }
+
+export interface LogData {
+  key: string;
+  value: string | number;
+}
+
+export class DataLoggingSensor extends Sensor {
```
Comment (microbit-matt-hillsdon) [inline src/board/sensors.ts:132]: Similarly to radio, I'm unsure whether this fits the sensor model. Let's see how radio works out first.

File: src/jshal.js [hunk 3]
```diff
@@ -215,4 +215,31 @@ mergeInto(LibraryManager.library, {
...
+  },
+
+  mp_js_hal_log_set_timestamp: function (period) {
+    board.dataLogging.setTimestamp(period);
```
Comment (microbit-matt-hillsdon) [inline src/jshal.js:228]: Should there be some kind of conversion here?

File: src/microbithal_js.c [hunk 4]
```diff
@@ -400,33 +403,33 @@ void microbit_hal_log_set_timestamp(int period) {
...
 int microbit_hal_log_data(const char *key, const char *value) {
+    return mp_js_hal_log_data(key, value);
     /*
     if (uBit.log.logData(key, value) != DEVICE_OK) {
         return -1;
     }
     return 0;
     */
-    return 0;
 }
```
Comment (microbit-matt-hillsdon) [inline src/microbithal_js.c:433]: To get the right "log full" behaviour I think we'll need to be propagating errors here.

File: src/jshal.js (hunk 1 above)
Comment (microbit-robert) [reply src/jshal.js]: Nope. [`26e9c8b`](https://github.com/microbit-foundation/micropython-microbit-v2-simulator/pull/36/commits/26e9c8b1d85e7dd05ae5ddc100b50d09d1e7c93b).

**Comment by microbit-matt-hillsdon** [comment]
Reworked into #41 after the API change. Closing.

**Pull request closed without merging**